### PR TITLE
feat(ci): add GHA for baremetal deployment

### PR DESCRIPTION
**Pull request by haikoschol**
## Changes

Add a GitHub Action for dispatching deployment on baremetal nodes to the infra repository.

### CODE REVIEW

**Review by dimartiro — APPROVED** [review]
I have not too much experience on this but LGTM.
Is there any way to test it before merge?

**Comment by haikoschol** [comment]
> Is there any way to test it before merge?

According to [the documentation](https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/manually-running-a-workflow):

> To trigger the workflow_dispatch event, your workflow must be in the default branch.

It might be possible to use a different event type, like `on: push`, but I'm not sure how to provide the required inputs then. And it still wouldn't test exactly this workflow. I think we just have to merge and fix if necessary.

File: .github/workflows/dispatch-baremetal.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          - snapshot 
+          - genesis
+      chain:
+        description: 'Chain'
+        required: true
+        default: 'westend'
+        type: choice
+        options:
+          - paseo
+          - westend
```
Comment (EclesioMeloJunior) [inline .github/workflows/dispatch-baremetal.yml:24]: can we add polkadot here as well?

**Review by EclesioMeloJunior — APPROVED** [review]

File: .github/workflows/dispatch-baremetal.yml (hunk 1 above)
Comment (haikoschol) [reply .github/workflows/dispatch-baremetal.yml:24]: We don't have a node for Polkadot yet.

**Pull request merged**